weldx/visualization.py

```diff
@@ -83,3 +324,1128 @@ def set_axes_equal(axes):
+        one will be created.
+    title : str
+        The title of the plot
+    limits :  Tuple[float, float] or List[Tuple[float, float]]
+        Each tuple marks lower and upper boundary of the x, y and z axis. If only a
+        single tuple is passed, the boundaries are used for all axis. If `None`
+        is provided, the axis are adjusted to be of equal length.
+    time_index : int
+        Index of a specific time step that should be plotted if the corresponding
+        coordinate system is time dependent
+    legend_pos : str
+        A string that specifies the position of the legend. See the matplotlib
+        documentation for further details
+
+    Returns
+    -------
+    matplotlib.axes.Axes :
+        The axes object that was used as canvas for the plot
+
+    """
+    if axes is None:
+        _, axes = new_3d_figure_and_axes()
+
+    for lcs, kwargs in cs_data:
+        if "time_index" not in kwargs:
+            kwargs["time_index"] = time_index
+        lcs.plot(axes, **kwargs)
+
+    _set_limits_matplotlib(axes, limits)
+
+    if title is not None:
+        axes.set_title(title)
+    axes.legend(loc=legend_pos)
+
+    return axes
+
+
+# k3d ----------------------------------------------------------------------------------
+
+
+class CoordinateSystemVisualizerK3D:
+    """Visualizes a `weldx.transformations.LocalCoordinateSystem` using k3d."""
+
+    def __init__(
+        self,
+        lcs,
+        plot: k3d.Plot = None,
+        name: str = None,
+        color: int = 0x000000,
+        show_origin=True,
+        show_trace=True,
+        show_vectors=True,
+    ):
+        """Create a `CoordinateSystemVisualizerK3D`.
+
+        Parameters
+        ----------
+        lcs : weldx.transformations.LocalCoordinateSystem
+            Coordinate system that should be visualized
+        plot : k3d.Plot
+            A k3d plotting widget.
+        name : str
+            Name of the coordinate system
+        color : int
+            The RGB color of the coordinate system (affects trace and label) as a 24 bit
+            integer value.
+        show_origin : bool
+            If `True`, the origin of the coordinate system will be highlighted in the
+            color passed as another parameter
+        show_trace :
+            If `True`, the trace of a time dependent coordinate system will be
+            visualized in the color passed as another parameter
+        show_vectors : bool
+            If `True`, the the coordinate axes of the coordinate system are visualized
+
+        """
+        coordinates, orientation = self._get_coordinates_and_orientation(lcs)
+        self._lcs = lcs
+        self._color = color
+
+        self._vectors = k3d.vectors(
+            origins=[coordinates for _ in range(3)],
+            vectors=orientation.transpose(),
+            colors=[[0xFF0000, 0xFF0000], [0x00FF00, 0x00FF00], [0x0000FF, 0x0000FF]],
```

Comment:
what do you think about defining and using constants like `RGB_RED=0xFF0000` in `visualization.py` ?

Reply:
Seems reasonable. Will do that.